build.gradle

```diff
@@ -132,11 +132,15 @@ check.dependsOn 'spotlessApply'
 def Task getDockerBuildTask(String artifactName, String projectDir) {
     return task ("buildDockerImage-$artifactName"(type: DockerBuildImage) {
         def buildTag = System.getenv('VERSION') ?: 'dev'
-        def buildPlatform = System.getenv('DOCKER_BUILD_PLATFORM') ?: 'linux/amd64'
-        def alpineImage = System.getenv('ALPINE_IMAGE') ?: 'alpine:3.4'
-        def postgresImage = System.getenv('POSTGRES_IMAGE') ?: 'postgres:13-alpine'
         def jdkVersion = System.getenv('JDK_VERSION') ?: '17.0.1'
-        def buildArch = System.getenv('DOCKER_BUILD_ARCH') ?: 'amd64'
+
+        def arch = System.getProperty("os.arch").toLowerCase()
+        def isArm64 = arch == "aarch64" || arch == "arm64"
+
+        def buildPlatform = System.getenv('DOCKER_BUILD_PLATFORM') ?: isArm64 ? 'linux/arm64' : 'linux/amd64'
+        def alpineImage = System.getenv('ALPINE_IMAGE') ?: isArm64 ? 'arm64v8/alpine:3.14' : 'alpine:3.14'
```

Comment:
there is also an [amd64/alpine:3.14](https://hub.docker.com/r/amd64/alpine) image.  What do you think about using that for the `amd64` arch builds instead of just `alpine:3.14`?

I bring this up because the `alpine` tag is shared by multiple images built for different architectures, so there is an edge case where you could have built an image for the arm64 arch using the arm version of `alpine:3.14` as a base image, but then when you try to build for the amd64 arch next, you will try to use the arm version of `alpine:3.14` as the base image to build amd64 images (because that is the one in your local machine's docker image cache).  Using `amd64/alpine` for amd64 should avoid this issue.

I get that if our users are only building this airbyte project this case wouldn't come up, but I think it could be very possible that users are building other images that simply use `alpine` as a base image, and I think users would normally expect the `alpine` image tag to refer to the version that matches the arch of their local machine.

Reply:
Good idea, I added these.